File: CSharp/Solutions/LongestCommonPrefix.cs
namespace Solutions;

public class LongestCommonPrefix
{
    public static string LongestCommonPrefix1(string[] strs)
    {
        string prefix = strs.First();

        for (int i = 1; i < strs.Length; i++)
        {
            while (!strs[i].StartsWith(prefix, StringComparison.Ordinal))
            {
                prefix = prefix[..(prefix.Length - 1)];
                if (prefix.Length == 0)
                {
                    return "";
                }
            }
        }

        return prefix;
    }
}

File: Go/longest_common_prefix/longest_common_prefix.go
package longest_common_prefix

import "strings"

func longestCommonPrefix(strs []string) string {
	longestPrefix := (func(strs []string) string {
		if len(strs) == 0 {
			return ""
		}
		return strs[0]
	})(strs)

	for i := 1; i < len(strs); i++ {
		for strings.Index(strs[i], longestPrefix) != 0 {
			longestPrefix = longestPrefix[:len(longestPrefix)-1]
			if longestPrefix == "" {
				return ""
			}
		}
	}

	return longestPrefix
}
